File: frontend/src/api/permission.ts
import api from './index'

export interface Permission {
  id: number
  name: string
  code: string
  type: 'menu' | 'button'
  parentId: number | null
  path: string
  icon: string
  sort: number
  status: 'active' | 'inactive'
  createdAt: string
}

export interface PermissionForm {
  id?: number
  name: string
  code: string
  type: 'menu' | 'button'
  parentId: number | null
  path: string
  icon: string
  sort: number
  status: 'active' | 'inactive'
}

export interface PermissionTree extends Permission {
  children: PermissionTree[]
}

// 获取权限列表
export function getPermissionList(params?: { name?: string }) {
  return api.get('/permission/list', { params })
}

// 获取权限树
export function getPermissionTree() {
  return api.get('/permission/tree')
}

// 创建权限
export function createPermission(data: PermissionForm) {
  return api.post('/permission/create', data)
}

// 更新权限
export function updatePermission(id: number, data: PermissionForm) {
  return api.put(`/permission/update/${id}`, data)
}

// 删除权限
export function deletePermission(id: number) {
  return api.delete(`/permission/delete/${id}`)
}

File: backend/src/main/java/com/vueadmin/entity/Permission.java
package com.vueadmin.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "bas_permission")
public class Permission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(unique = true, nullable = false, length = 50)
    private String code;

    @Enumerated(EnumType.STRING)
    @Column(length = 10)
    private Type type = Type.menu;

    @Column(name = "parent_id")
    private Long parentId;

    @Column(length = 100)
    private String path;

    @Column(length = 50)
    private String icon;

    @Column(columnDefinition = "INT DEFAULT 0")
    private Integer sort = 0;

    @Enumerated(EnumType.STRING)
    @Column(length = 10)
    private Status status = Status.active;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum Type {
        menu, button
    }

    public enum Status {
        active, inactive
    }

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
